packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -675,98 +662,101 @@ - (NSDictionary *)encodingMap
 
 #pragma mark - Internal methods
 
-- (void)_downloadResumableCreateSessionWithUrl:(NSURL *)url withScopedPath:(NSString *)scopedPath withUUID:(NSString *)uuid withOptions:(NSDictionary *)options withResumeData:(NSData * _Nullable)resumeData withResolver:(UMPromiseResolveBlock)resolve withRejecter:(UMPromiseRejectBlock)reject
+- (NSURLSession *)_createSession:(id<NSURLSessionDelegate>)delegate withOptions:(NSDictionary *)options
 {
-  NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
+  EXFileSystemSessionType sessionType = [options[@"sessionType"] intValue] ?: EXFileSystemBackgroundSession;
+  NSDictionary *headers = options[@"headers"];
```

Comment:
Should we verify that the `headers` really are a dictionary of `<NSString *, NSString *>`? Will the app crash if it's not?